api-documentation/shacl.md

```diff
@@ -0,0 +1,181 @@
+# Using SHACL to describe data structures
+
+## Story
+
+As an API publisher
+
+I want to use SHACL to describe data structures
+
+So that consumers can use existing tooling to consume the API
+
+## Details
+
+SHACL, which stands for Shapes Constraint Language is an RDF vocabulary, defined in a [W3C recommendation][shacl], which is used to describe graph shapes.
+
+It is very expressive, more rich than `hydra:Class` et.al., and out of the box provides multiple features to provide an accurate description of request payloads.
+
+[Many uses for SHACL have been proposed][ucr] such as, but not limited to, validation and building user interfaces dynamically.
+
+This page shows how Shapes can be used to replace or extend Hydra Core vocabulary terms within the API Documentation and resource representations.
+
+[shacl]: https://www.w3.org/TR/shacl/
+[ucr]: https://www.w3.org/TR/shacl-ucr/
+
+### Announce SHACL in API Documentation
+
+An API which chooses to use SHACL in addition to the built-in `hydra:Class` `MUST` announce that fact in its API Documentation resource
+
+```json
+{
+  "@context": {
+    "hydra": "http://www.w3.org/ns/hydra/core#",
+    "requires": { "@id": "hydra:requires", "@type": "@id" }
+  },
+  "@id": "/api",
+  "@type": "hydra:ApiDocumentation",
+  "requires": "http://www.w3.org/ns/hydra/shacl"
+}
+```
+
+By adding the `requires` property with the `hydra/shacl` identifier the client will be prepared to find SHACL terms within the payloads coming from this API.
+
+### Describing request bodies
+
+The most obvious use for Shapes is to use them to annotate supported operations' payloads. They will simply be added alongside `hydra:Class` as objects of `hydra:expects`.
+
+```json
+{
+  "@context": "/api/context.jsonld",
+  "@id": "/api",
+  "@type": "hydra:ApiDocumentation",
+  "supportedClass": [{
+    "@id": "mov:Movies",
+    "supportedOperation": {
+      "@type": "schema:CreateAction",
+      "method": "POST",
+      "expects": ["mov:Movie", "mov:NewMovieShape"]
+    }
+  }, {
+    "@id": "mov:Movie",
+    "@type": "Class",
+    "supportedProperty": [{
+      "property": "schema:name"
+    }, {
+      "property": "mov:director"
+    }, {
+      "property": "schema:genre"
+    }]
+  }]
+}
+```
+
+By adding a second value to the `expects` annotation, the client will discover that the payload of this operation is expected to be not only a representation of a `mov:Movie` but also that the request body has to conform to the given SHACL Shape.
+
+{% hint style="info" %}
+Client which do not understand SHACL will ignore it and proceed with only the `hydra:Class` description.
```

Comment:
What if client understands both. Which one takes precedence?

Reply:
As answered above; no precedence; the `Shape` should complement `Class`. The Shape provides more details about a resource representation of `mov:Movie`. If a client does not understand SHACL, they would still have the limited, but sufficient, Hydra Core description